render_pdf/src/make_latex.py

```diff
@@ -0,0 +1,80 @@
+from typing import Union
+import parser
+import os
+import jinja2
+
+# TODO probably best to use pydantic for a config template for the dict below. 
+data = {
+        'language' : "sanskrit",
+        'title' : "",
+        'author' : "",
+        'body' : [],
+        'analysis': []
+    }
+
+latex_jinja_env = jinja2.Environment(
+        block_start_string = '<*',
+        block_end_string = '*>',
+        variable_start_string = '<<',
+        variable_end_string = '>>',
+        comment_start_string = '<=',
+        comment_end_string = '=>',
+        loader = jinja2.FileSystemLoader(os.path.abspath('.'))
+    )
+template = latex_jinja_env.get_template('template.tex')
+
+## TODO add an argument for receiving language, default is sanskrit + devanagari
+
+def generate_fields(link: str, lipi: Union["SKT","IAST"], author: str) -> tuple[list, str, str]:
+    # TODO handle lipi input
+    _ = lipi
+    #####
+
+    verses, analysis, title, parsed_author = parser.parse(url=link)
+    if parsed_author != "":
+        author = parsed_author
+    parsed_verses = []
+    parsed_analysis = []
+    
+    for verse in verses:
+        for line in verses[verse]:
+            l = [verse]
+            for segment in verses[verse][line]:
+                text = verses[verse][line][segment]
+                l.append(text)
+            parsed_verses.append(l)
+    for verse in analysis:
+        for line in verses[verse]:
+            l = [verse]
+            for segment in verses[verse][line]:
+                text = verses[verse][line][segment]
+                l.append(text)
+            parsed_analysis.append(l)
+
+    return parsed_verses, parsed_analysis, title, author
+
+def generate_texfile(link, lipi, author):
+    # TODO Author can either be passed in or extracted. Some texts don't have this. 
+    # Need to standardize input texts. 
+    body, analysis, title, author = generate_fields(link, lipi, author)
+    global data
+    data['body'] = body
+    data['title'] = title.lower()
+    data['author'] = author.lower()
+    data['analysis'] = analysis
+    if analysis == []:
+        data['hasAnalysis'] = False
```

Comment:
In Python, it's idiomatic to check boolean status against the collection correctly, thus `if analysis` vs `if hasAnalysis`

Reply:
Simpler to write data['hasAnalysis'] = not analysis; does the same thing. But I can do:
॑if not analysis:
    data['hasAnalysis'] = False
॑

Reply:
what I'm saying is that `hasAnalysis` is redundant because wherevere you would use `if data['hasAnalysis']` you could instead just write `if analysis`